Cython/Compiler/ModuleNode.py

```diff
@@ -528,16 +528,19 @@ def _serialize_lineno_map(self, env, ccodewriter):
         markers = ccodewriter.buffer.allmarkers()
 
         d = defaultdict(list)
-        for c_lineno, cython_lineno in enumerate(markers):
-            if cython_lineno > 0:
-                d[cython_lineno].append(c_lineno + 1)
+        for c_lineno, cython_tup in enumerate(markers):
+            cython_fp = cython_tup[0]
+            cython_lineno = cython_tup[1]
+            if cython_lineno > 0 and cython_fp.filename is not None:
+                d[cython_fp, cython_lineno].append(c_lineno + 1)
```

Comment:
```suggestion
        for c_lineno, (src_desc, src_lineno) in enumerate(markers):
            if src_lineno > 0 and src_desc.filename is not None:
                d[(src_desc.filename, src_lineno)].append(c_lineno + 1)
```
and then adapt the code below accordingly. `cython_fp` doesn't seem a very clear name to me.